src/runtime_src/core/tools/xbreplay/src/main.cpp

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+#include "seq_reconstructor.hpp"
+#include "utils/cmd_args.hpp"
+#include "utils/logger.hpp"
+
+#include <iostream>
+#include <memory>
+namespace {
+using namespace xrt::tools::xbreplay;
+
+class successfull_termination : public std::exception
+{
+  public:
+  const char* what() const noexcept override
+  {
+    return "Program terminated successfully";
+  }
+};
+
+/*
+ * This function is used to parse the command line arguments
+ */
+static std::pair<std::string, std::string>
+parse_command_line_arguments(std::vector<std::string>& cmd_params)
+{
+  std::string trace_file;
+  std::string mem_file;
+  std::vector<std::string>& args = cmd_params;
+  cmd_args_opt opt;
```

Comment:
In C++ we never pre-define variables.  Move declaration and definition to where the variables are used.